staging/src/k8s.io/apiserver/pkg/cel/library/quantity.go

```diff
@@ -254,127 +237,73 @@ func quantityGetValue(arg ref.Val) ref.Val {
 }
 
 func quantityGetSign(arg ref.Val) ref.Val {
-	q, ok := arg.Value().(*resource.Quantity)
-	if !ok {
-		return types.MaybeNoSuchOverloadErr(arg)
-	}
+	q := arg.(apiservercel.Quantity)
 	return types.Int(q.Sign())
 }
 
 func quantityIsGreaterThan(arg ref.Val, other ref.Val) ref.Val {
-	q, ok := arg.Value().(*resource.Quantity)
-	if !ok {
-		return types.MaybeNoSuchOverloadErr(arg)
-	}
-
-	q2, ok := other.Value().(*resource.Quantity)
-	if !ok {
-		return types.MaybeNoSuchOverloadErr(arg)
-	}
+	q := arg.(apiservercel.Quantity)
+	q2 := other.(apiservercel.Quantity)
 
-	return types.Bool(q.Cmp(*q2) == 1)
+	return types.Bool(q.Cmp(*q2.Quantity) == 1)
 }
 
 func quantityIsLessThan(arg ref.Val, other ref.Val) ref.Val {
-	q, ok := arg.Value().(*resource.Quantity)
-	if !ok {
-		return types.MaybeNoSuchOverloadErr(arg)
-	}
+	q := arg.(apiservercel.Quantity)
+	q2 := other.(apiservercel.Quantity)
 
-	q2, ok := other.Value().(*resource.Quantity)
-	if !ok {
-		return types.MaybeNoSuchOverloadErr(arg)
-	}
-
-	return types.Bool(q.Cmp(*q2) == -1)
+	return types.Bool(q.Cmp(*q2.Quantity) == -1)
 }
 
 func quantityCompareTo(arg ref.Val, other ref.Val) ref.Val {
-	q, ok := arg.Value().(*resource.Quantity)
-	if !ok {
-		return types.MaybeNoSuchOverloadErr(arg)
-	}
+	q := arg.(apiservercel.Quantity)
```

Comment:
Could you help me understand the type-shift? Is one the CEL type (`apiservercel.Quantity`) and the other the internal type?

Reply:
Sure thing! I did this because the associated overload declarations https://github.com/kubernetes/kubernetes/blob/bf7fcfdc7f1c0aac000e4edf38eaa59b3ce941d4/staging/src/k8s.io/apiserver/pkg/cel/library/quantity.go#L156-L184 set `apiservercel.Quantity` as the arg/result type. So I felt like this was the "safe" cast that would remove the need for type assertions, even though https://github.com/kubernetes/kubernetes/blob/bf7fcfdc7f1c0aac000e4edf38eaa59b3ce941d4/staging/src/k8s.io/apiserver/pkg/cel/quantity.go#L36-L39 `*resource.Quantity` is currently an embedded field of `apiservercel.Quantity`